atomic_reactor/utils/resource_manager.py

```diff
@@ -18,43 +18,59 @@
 from shlex import quote
 from typing import List
 
-RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
+RESOURCE_LOCK_SCHEMA = "schemas/resource-lock-content.json"
 SSH_COMMAND_TIMEOUT = 45
 TIMEOUT_FAIL_BUILD = 3000
-RELATIVE_PATH = 'resource_manager'
-RESOURCE_INFO_JSON = 'info.json'
-RETRY_ON_SSH_EXCEPTIONS = (paramiko.ssh_exception.NoValidConnectionsError,
-                           paramiko.ssh_exception.SSHException)
+RELATIVE_PATH = "resource_manager"
+RESOURCE_INFO_JSON = "info.json"
+RETRY_ON_SSH_EXCEPTIONS = (
+    paramiko.ssh_exception.NoValidConnectionsError,
+    paramiko.ssh_exception.SSHException,
+)
 BACKOFF_FACTOR = 5
 MAX_RETRIES = 3
 
-logger = logging.getLogger(__name__)
+LOGGER = logging.getLogger(__name__)
 
 
 class ResourceManagerError(RuntimeError):
-    """ Base module exception """
+    """Base module exception"""
 
 
 class LockError(ResourceManagerError):
-    """ Attempt for resource locking returned non-zero exitcode """
+    """Attempt for resource locking returned non-zero exitcode"""
 
 
 class LockRetry(ResourceManagerError):
-    """ Retry locking, no resources were available """
+    """Retry locking, no resources were available"""
 
 
-class RemoteHost:
+class RemoteHostLoggerAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        """Manipulate the message to always prepend hostname"""
+        msg, kwargs = super().process(msg, kwargs)
+        hostname = self.extra.get("hostname")
+        if hostname:
+            msg = f"{hostname}: {msg}"
+        return msg, kwargs
+
 
-    def __init__(self, *, hostname: str, username: str, ssh_keyfile: str):
+class RemoteHost:
+    def __init__(
+        self, *, hostname: str, username: str, ssh_keyfile: str, logger: logging.Logger = LOGGER
+    ):
         """
         :param hostname: str, remote hostname for ssh connection
         :param username: str, username for ssh connection
         :param ssh_keyfile: str, filepath to ssh private key
+        :param logger: logging.Logger, logger to use. Optional, defaults to LOGGER
         """
         self._hostname = hostname
         self._username = username
         self._ssh_keyfile = ssh_keyfile
 
+        self.logger = RemoteHostLoggerAdapter(logger, {"hostname": hostname})
```

Comment:
Can we have per class specific logger. Passing logger via argument is not consistent with other modules (and I think we don't need it)
```suggestion
        self.logger = RemoteHostLoggerAdapter(
            logging.getLogger(f"{self.__class__.__module__}.{self.__class__.__name__}"),
             {"hostname": hostname})
```

Reply:
or just keep module logger there without param as is done in other modules in OSBS